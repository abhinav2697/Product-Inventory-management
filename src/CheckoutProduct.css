.checkoutProduct {
    margin: 80px 20px;
    display: flex;
}

.checkoutProduct__image {
    object-fit: contain;
    width: 250px;
    height: 250px;
    margin-right: 50px;
}

.checkoutProduct__info {
    height: 100px;
    margin-top: 30px;
    margin-bottom: 5px;
    text-align: center;
    align-items: center;
}
 


.checkoutProduct__title {
    font-size: 20px;
   font-weight: bold;
}

.checkoutProduct__rating {
    display: flex;
}

.checkProduct__description{
    font-weight: bold;
    font-size: 15px;
    
}
.checkoutProduct__price {
    font-size: 18px;
}

.checkProduct__manufacturer{
    font-weight: bold;
    font-size: 15px;
}
.checkProduct__quantity{
    font-weight: bold;
    font-size: 15px;
}


.checkoutProduct__info > button {
    margin-top: 20px; 
    background-color: #E5E6E7;
    color: #111111;
    height: 40px;
    width: 180px;
    font-size: 14px;
    font-weight: 700;
    border: none;
    border-radius: 30px;
    cursor: pointer;
}
.checkoutProduct__info > button:hover {
    background-color: yellow;
}
@media (max-width:670px)  {
    .checkoutProduct {
        flex-direction: row;
        margin: 80px 0px;
    }
    .checkoutProduct__image {
        /* margin-right: 0px; */
        padding: 0px;
        margin: 0px;
        width: 200px;
        height: 200px;
    }
    .checkoutProduct__info {
        padding-left: 20px;
        text-align: center;
    }
    .checkoutProduct__title {
        font-size: 17px;
    }
    .checkoutProduct__price {
        font-size: 15px;
    }
    .checkoutProduct__rating {
        justify-content: center;
    }
    .checkoutProduct__info > button {
        font-size: 13px;
    }
}