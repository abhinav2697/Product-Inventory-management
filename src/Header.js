import React ,{useState,useEffect}from 'react';
import './Header.css';
import { Link ,useHistory} from 'react-router-dom';
import SearchIcon from '@material-ui/icons/Search';
import LocalShippingIcon from '@material-ui/icons/LocalShipping';
import PersonOutlineIcon from '@material-ui/icons/PersonOutline';
import ShoppingBasketIcon from '@material-ui/icons/ShoppingBasket';
import MenuIcon from '@material-ui/icons/Menu';
import Button from '@material-ui/core/Button';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';

import * as ProductActions from '../src/actions/ProductActions'

import { connect } from "react-redux";
import TextField from '@material-ui/core/TextField';
import Autocomplete from '@material-ui/lab/Autocomplete';



export function Header(props) {
    const [anchorEl, setAnchorEl] = React.useState(null);
    const [searchTerm,setSearchTerm] = useState('');
    const [searchResults, setSearchResults] = useState([]);
    function fetchAllProducts(){
        props.getProducts()
      }
      useEffect(() => {
        fetchAllProducts();
      },[]);
    
    const handleChange = event => {
        const searchedTerm = event.target.value;
        setSearchTerm(searchedTerm)
        if(!searchedTerm){
            console.log('No Search Term')
        }
        const searchProducts = props.products.filter((product)=> product.ProductName == searchedTerm)
        //console.log(searchProducts)
        if(searchProducts.length != 0){
        props.setFilteredProducts(true,searchProducts);
        }
    };
   
    
    


  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };

  let user=localStorage.getItem('user-info')
  const history=useHistory();
  function logout(){
      props.logout();
      setTimeout(() => {
        history.push("/login");
        }, 500); 
  }
       
    return (
        <nav className="header">
            <div className="header__right">
                <MenuIcon className="header__menu"/>
            </div>
                <Link to="/">
                    <img 
                        className="header__logo" 
                        src="https://cdn.shippify.co/dash/users/img/users-shippify-logo.svg"
                        alt=""
                    />
                </Link>
                <Link to="/products" className="header__pro">
                    <h4 className="header__title">Products</h4>
                </Link>
                
                
                
            <div className="header__search">
            {/* <SearchIcon className="header__searchIcon" /> */}
            <Autocomplete className="header__searchInput"
            
                options={props.products.map((product) => product.ProductName)}
                renderInput={(products) => (
                <TextField {...products}  type="text"  value={searchTerm} 
                onChange={handleChange} label="What are you looking for?" margin="normal" variant="outlined" />
                )}
                />
                {/* <input 
                    type="text" 
                    className="header__searchInput"
                    placeholder="What are you looking for?"   
                    value={searchTerm} 
                    onChange={handleChange}
                /> */}
                 
                
                    
                
            </div>

            <div className="header__nav">
                <Link to="/" className="header__link">
                    <div className="header__option">
                    <LocalShippingIcon/>
                    </div>
                </Link>
                    {
                        props.isLoggedIn ? (
                            <div  className="header__optionProfile">
                            <Button aria-controls="long-menu" aria-haspopup="true" onClick={handleClick}>
                            <PersonOutlineIcon></PersonOutlineIcon>
                            </Button>
                        <Menu
                                id="long-menu"
                                anchorEl={anchorEl}
                                keepMounted
                                open={Boolean(anchorEl)}
                                onClose={handleClose}
                             >
                            <MenuItem onClick={handleClose}>
                                <Link to="/admin">
                                    Admin
                                </Link>
                            </MenuItem>
                            <MenuItem onClick={handleClose}>
                                <Link onClick={logout}>Logout</Link>
                            </MenuItem>
                        </Menu>
                        </div>
                        ) : (
                            <Link to="/login" className="header__link">
                                <span className="header__login">
                                    Login
                                </span>
                            </Link>
                        )
                    }
                <Link to="/checkout" className="header__link">
                    <div className="header__optionBasket">
                        <ShoppingBasketIcon/>
                        <span className="header__optionLineTwo header__basektCount">
                            {props.basket?.length}
                        </span>
                    </div>
                </Link>
                
                
                
            </div>
        </nav>
    )
}

const getDataFromStore = (state) =>{
return {
    basket: state.userReducer.basket,
    isLoggedIn: state.productReducer.isLoggedIn,
    products: state.productReducer.allProducts
}
}

const mapReducerToStore = (dispatch) => {
    return {
      logout: () =>
        dispatch(ProductActions.logout()),
        getProducts: () => {
            dispatch(ProductActions.getAllProducts());
          },
          setFilteredProducts:(searchDetails,searchProducts)=>{
              dispatch(ProductActions.Search(searchDetails,searchProducts));
          }
    };
  };

export default connect(getDataFromStore,mapReducerToStore)(Header);
