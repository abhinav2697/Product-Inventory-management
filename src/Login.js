import React from 'react';
import './Login.css'
import { Formik, Form,Field } from 'formik';
import { Link, useHistory } from "react-router-dom";
import * as Yup from 'yup';
import {connect} from "react-redux";
import * as ProductActions from '../src/actions/ProductActions'


const validateSchema = Yup.object().shape({
    email: Yup.string().email('Email is invalid').required('Email is required'),
    password: Yup.string().min(6, 'Password must be at least 6 charaters').required('Password is required'),
})


export function Login(props){
    let history=useHistory();
    const addLoginForm = async(values)=>{
        const data={
           email:values.email,
           password:values.password
        }
        props.addLogin(data);
       setTimeout(() => {
        history.push("/products");
        }, 500); 
    }

  return (
      <div className='login'>
          <Link to='/'>
          <img  className="login__logo"
                src="https://cdn.shippify.co/dash/users/img/users-shippify-logo.svg"
                alt=""
            />
            </Link>
            <div className='login__container'>
    <Formik
      initialValues={{
        email: '',
        password: '',
      }}
      validationSchema={validateSchema}
      onSubmit={(values)=> addLoginForm(values)}
    >
      {({errors,touched})=>(
        <React.Fragment>
            <Form > 
                <h2>Login Form</h2>
                <Field name='email' placeholder='Enter Email' type='email' /><br/>
                {errors.email && touched.email? <div style ={{color:'red'}}>{errors.email}</div> : null}
                <Field name='password' placeholder='Enter Password' type='password'  /><br/>
                {errors.password && touched.password ? <div style ={{color:'red'}}>{errors.password}</div> : null}
                <button type='submit' class='login__signInButton'>Login</button>
                <a href="/SignUp">Do not have the account?  Register Here</a>
            </Form>
            </React.Fragment>
        )}
    
    </Formik>
    </div>
    </div>
    
  )
}

const mapReducerToStore = (dispatch) => {
    return {
      addLogin: (loginDetails) =>
        dispatch(ProductActions.addLogin(loginDetails)),
    };
  };
  
export default connect(null, mapReducerToStore)(Login);