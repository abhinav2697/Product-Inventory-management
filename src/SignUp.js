import React from 'react';
import './SignUp.css'
import {Formik,Form,Field} from 'formik';
import * as yup from 'yup';
import {Link} from 'react-router-dom'
import {useHistory} from 'react-router-dom';
import {v4 as uuId} from 'uuid';
import {connect} from "react-redux";
import * as ProductActions from '../src/actions/ProductActions'

const validateSchema = yup.object().shape({
  email: yup.string().email('Email is invalid').required('Email is required'),
  password: yup.string().min(6, 'Password must be at least 6 charaters').required('Password is required'),
  firstname: yup.string().required('First Name is Required'),
  lastname: yup.string().required('Last Name is Required'),
  location: yup.string().required('Location is Required'),
  mobilenumber: yup.number().required('Mobile Number is Required'),

})
export function SignUp(props){
  const emailId=uuId();
  let history=useHistory();
  const addSignUpForm = async(values)=>{
      const data={
          id:emailId,
          email:values.email,
          password:values.password,
          firstname:values.firstname,
          location:values.location,
          mobilenumber:values.mobilenumber
      }
      props.addSignUp(data);
     setTimeout(() => {
      history.push("/login");
      }, 500); 
  }

  
  return (
      <div className='signup'> 
            <Link to='/'>
                <img
                    className="signup__logo"
                    src="https://cdn.shippify.co/dash/users/img/users-shippify-logo.svg"
                />
            </Link>

          <div className='signup__container'>
          <Formik
    initialValues ={{
      email:'',
      password:'',
      firstname:'',
      location:'',
      mobilenumber:'',
    }}
    
    validationSchema={validateSchema}
    onSubmit={(values)=> addSignUpForm(values)}
    
    >
        {({errors,touched})=>(
            <Form > 
                <h2>Sign Up</h2>
                <Field name='email' placeholder='Enter Email' type='email' /><br/>
                {errors.email && touched.email ? <div style ={{color:'red'}}>{errors.email}</div> : null}
                <Field name='password' placeholder='Enter Password' type='password' /><br/>
                {errors.password && touched.password ? <div style ={{color:'red'}}>{errors.password}</div> : null}
                <Field name='firstname' placeholder='Enter First Name' type='text' /><br/>
                {errors.firstname && touched.firstname ? <div style ={{color:'red'}}>{errors.firstname}</div> : null}
                <Field name='lastname' placeholder='Enter Last Name' type='text' /><br/>
                {errors.lastname && touched.lastname ? <div style ={{color:'red'}}>{errors.lastname}</div> : null}
                <Field name='location' placeholder='Enter Location' type='text'  /><br/>
                {errors.location && touched.location? <div style ={{color:'red'}}>{errors.location}</div> : null}
                <Field name='mobilenumber' placeholder='Enter Mobile Number' type='number'  /><br/>
                {errors.mobilenumber && touched.mobilenumber ? <div style ={{color:'red'}}>{errors.mobilenumber}</div> : null}
                <button type='submit' class='signup__registerButton'> SignUp</button>
                <a href="/Login">already have an account?  login here </a>
            </Form>
        )}
    </Formik>
    </div>
    </div>
    )
 
}


const mapReducerToStore = (dispatch) => {
  return {
    addSignUp: (signupDetails) =>
      dispatch(ProductActions.addSignUp(signupDetails)),
  };
};

export default connect(null, mapReducerToStore)(SignUp);
