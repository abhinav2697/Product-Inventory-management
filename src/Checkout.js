import React from 'react';
import CheckoutProduct from './CheckoutProduct';
import Subtotal from './Subtotal';
import { Link } from 'react-router-dom';
import LockIcon from '@material-ui/icons/Lock';
import LoyaltyIcon from '@material-ui/icons/Loyalty';

import './Checkout.css';

import {connect} from react-redux;

export function Checkout(props) {     
    return (
        <div className="checkout">
            <div className="checkout__left">
                {props.basket?.length === 0 ? (
                    <div className="checkout__empty">
                        <img src="https://www.ikea.com/nl/en/shoppingcart/static/media/ill-alien.cb42647f.svg" alt=""/>
                        <h5>Shopping cart (0)</h5>
                        <div  className="checkout__emptyDetails">
                            <p>
                                Your shopping bag is currently empty
                            </p>
                            <Link to="/login" className="checkout__emptyLink">
                                Create an account
                            </Link>
                        </div>
                        
                        <div className="checkout__Redirect">
                            
                            <button>
                                <Link to="/products" className="checkout__buttonLink">
                                    Browse our products
                                </Link>
                            </button> 
                            <button>
                                <Link to="/login" className="checkout__buttonLink">
                                    Login
                                </Link>
                            </button>
                        </div>
                        <div className="checkout__emptyShow">
                            <div className="checkout__emptyExtra">
                                <LockIcon/>
                                <h5>Secure shopping</h5>
                                <p>with SSL data encryption</p>
                            </div>
                            <div className="checkout__emptyExtra">
                                <LoyaltyIcon/>
                                <h5>One year garantee</h5>
                                <p>to change your mind</p>
                            </div>
                        </div>
                        
                    </div>
                  
                ) : (
                    <div>
                        <h2 className="checkout__title">Your shopping Basket</h2>

                        {/* List out all of the checkout products */}
                        { 
                        
                        props.basket?.map((item) => ( 
                            
                             <CheckoutProduct 
                                 id={item.id}
                                 ProductName={item.ProductName}
                                 Description={item.ProductDescription}
                                 Price={item.Price}
                                 Manufacturer={item.Manufacturer}
                                 Quantity={item.Quantity}   
                            />
                            ))}
                    </div>
                )}
            </div>

            {props.basket.length > 0 && (
                <div className="checkout__right">
                    {/* <h2>Subtotal</h2> */}
                    <Subtotal/>
                </div>
            )}
        </div>
    );
}

const getDataFromUserStore = (state) =>{
    return {
        basket: state.userReducer.basket
    }
}

export default connect(getDataFromUserStore,null)(Checkout);
