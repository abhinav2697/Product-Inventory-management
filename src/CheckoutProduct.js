import React from 'react';
import './CheckoutProduct.css';
import { connect } from "react-redux";
import * as reducerActions from './reducerActions'

function CheckoutProduct(props) {
    // console.log(id, title,description,manufacturer, price,quantity);
    const removeFromBasket = () => {
        // remove item from basket ...
        props.removeFromBasket(props.id)
    }
    
    return (
        <div className="checkoutProduct">
         

            <div className="checkoutProduct__info">
                <p className="checkoutProduct__title">Name:{props.ProductName}</p>
                 <p className="checkProduct__description">Des:{props.ProductDescription}</p> 
                <p className="checkoutProduct__price">Price:
                    <small> ‎₹ </small>
                    <strong>{props.Price}</strong> 
                </p>
                <p className="checkProduct__manufacturer"><small>Manufacturer:{props.Manufacturer}</small></p>
                <p className="checkProduct__quantity"><small>Quantity:{props.Quantity}</small></p>

                
            
                <button onClick={removeFromBasket}>Remove from basket</button>
            </div>
        </div>
    );
}


const componentActions = (dispatch) =>{
    return {
            removeFromBasket: (productDetails) => dispatch(reducerActions.popFromBasket(productDetails))
    }
}

export default connect(null,componentActions)(CheckoutProduct);
