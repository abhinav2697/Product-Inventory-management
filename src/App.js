import React from 'react';
import {BrowserRouter as Router, Switch,Route} from "react-router-dom";
import AllProductsPage from './components/AllProductsPage';
import About from './components/About'
//import {Link } from 'react-router-dom'
import AddProduct from './components/AddProduct';
//import ProductView from './components/ProductView';
import Header from '../src/Header'
import Login from './Login'
import Home from './Home'
import Signup from './SignUp';
import Footer from './Footer'
import './App.css';
import Products from './Products';
import Checkout from './Checkout'
import Admin from '../src/components/Admin';


export function App() {
  return (
      <Router>
        <div class Name="app">
          <Switch>
          <Route path="/Checkout">
            <Header/>
            <Checkout />            
          </Route>

          <Route path="/login">
            <Header/>
            <Login/>
          </Route>
          <Route path="/SignUp">
            <Header/>
            <Signup/>
            
          </Route>
          <Route exact path="/about">
          <About/>
        </Route>
        <Route path="/products">
            <Header/>
            <Products/>
            <Footer/>
          </Route>

          <Route path="/Admin">
            <Header/>
            <Admin/>
            <Footer/>
          </Route>
          
        <Route exact path="/AddProduct">
          <Header/>
          <AddProduct/>
          <Footer/>
        </Route>

        
        <Route exact path="/AllProductsPage">
          <AllProductsPage/>
        </Route>
        <Route path="/">
            <Header/>
            <Home/>
            <Footer/>
          </Route>
        
          </Switch>
          </div>
          </Router> 
      //  {/* <Link to="/">About</Link>&emsp;  */}
      // <Link to="/products">Products</Link>&nbsp; 
      //  {/* <Route exact path="/" component={About} ></Route>  */}
      //  <Route exact path="/products" component={AllProductsPage}></Route> 
      // <Route exact path="/AddProduct" component={AddProduct}></Route>
      // <Route exact path="/ProductView/:ProductName" component={ProductView}>
        // {/* </Route>  */}
        
      
     


  
  );
}

export default App;
