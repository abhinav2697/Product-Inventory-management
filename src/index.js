import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {Provider} from 'react-redux'
import thunk from 'redux-thunk';


import { createStore, applyMiddleware, compose,combineReducers } from 'redux';
import ProductReducer from "./reducers/ProductReducer"
import reducer from './reducer';

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;


const combine_reducers = combineReducers({productReducer: ProductReducer,userReducer: reducer}) 

const store = createStore(combine_reducers,composeEnhancers(applyMiddleware(thunk)))

ReactDOM.render(
  <Provider store={store}>
    <App />
    
  </Provider>,
  document.getElementById('root')
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
