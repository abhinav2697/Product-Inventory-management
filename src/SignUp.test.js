import React from 'react';
import { shallow ,configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {SignUp} from '../src/SignUp';
import { Form } from 'react-bootstrap';
import { Field } from 'react-bootstrap';
import { Formik } from 'formik';

configure({ adapter: new Adapter() });
let wrapper;

describe('signup component tests', ()=> {
    wrapper = shallow(<SignUp />);


   const w1=shallow(<Form />);

   it('has a login button', () => {
    const wrapper = shallow(<SignUp/>);
    expect(wrapper.containsMatchingElement(<button type="submit">SignUp</button>)).toEqual(false);
  });

  it('has a email input field', () => {
    const wrapper = shallow(<SignUp />);
    expect(wrapper.containsMatchingElement(<input type="email" name='email'/>)).toEqual(false);
  });

  it('has a password input field', () => {
    const wrapper = shallow(<SignUp />);
    expect(wrapper.containsMatchingElement(<input type="password"  name="password"/>)).toEqual(false);
  });
 
  
 


    test('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    
    it('displays correct heading for Users', () => {
        const w2 = shallow(<SignUp />);
    expect(w2.equals(<div className="signup" />)).toEqual(false);
    })

    // it('displays correct heading for Users', () => {
    //     expect(wrapper.find("button")).simulate('click');
    // })
    
});