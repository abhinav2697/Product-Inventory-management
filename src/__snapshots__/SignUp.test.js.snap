// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signup component tests renders correctly 1`] = `ShallowWrapper {}`;
